### PR TITLE
[e2etest] Use OneDrive SDK for client directed MSA login test instead of Live SDK.

### PR DESCRIPTION
**Pull request by dhei**
[Live SDK](https://github.com/liveservices/LiveSDK) is going to be replaced by [OneDrive SDK](https://github.com/OneDrive/onedrive-sdk-csharp), so it's time to drop Live SDK and pick up OneDrive SDK in e2etest. **OneDrive SDK only supports Windows 8.1, WindowsPhone 8.1 and UWP.** 

This PR fixes the client directed login flow for Microsoft Account using OneDrive SDK. 

PS: OneDrive [wiki](https://github.com/OneDrive/onedrive-sdk-csharp/blob/master/docs/auth.md#microsoft-account-msa-authentication) explains it very well how to use it's authentication API.

@pragnagopa @brettsam Can you take a look?

### CODE REVIEW

**Comment by brettsam** [comment]
I can't remember -- is this the only client login test we have? I'd expect all of them to need to be on the UI thread.

Other than that, this looks good to me.

File: e2etest/WindowsStore.E2ETest/Functional/ClientSDKLoginTests.cs [hunk 1]
```diff
@@ -15,26 +18,35 @@ public class ClientSDKLoginTests : FunctionalTestBase
...
+                    accessToken["access_token"] = session.AccessToken;
 
-                    Log(string.Format("{0} is now logged into MobileService with userId - {1}", meResult.Result["first_name"], loginResult.UserId));
+                    MobileServiceUser user = await this.GetClient().LoginAsync(MobileServiceAuthenticationProvider.MicrosoftAccount, accessToken);
+
+                    Log(string.Format("Log in with Microsoft Account succeeded - userId {0}", user.UserId));
+                }
+                else
+                {
+                    Assert.Fail("Log in with Live SDK failed");
```
Comment (pragnagopa) [inline e2etest/WindowsStore.E2ETest/Functional/ClientSDKLoginTests.cs]: Nit: OneDrive SDK

File: e2etest/WindowsStore.E2ETest/Functional/ClientSDKLoginTests.cs [hunk 2]
```diff
@@ -15,26 +18,35 @@ public class ClientSDKLoginTests : FunctionalTestBase
...
                 }
             }
             catch (Exception exception)
```
Comment (pragnagopa) [inline e2etest/WindowsStore.E2ETest/Functional/ClientSDKLoginTests.cs]: Remove the try catch. Redundant as the test framework logs messages from exceptions

**Comment by pragnagopa** [comment]
:) :)

File: e2etest/WindowsStore.E2ETest/Functional/ClientSDKLoginTests.cs (hunk 2 above)
Comment (dhei) [inline e2etest/WindowsStore.E2ETest/Functional/ClientSDKLoginTests.cs]: @pragnagopa try-catch block is very handy for debugging and trouble shooting, so let's keep it this way.

**Comment by dhei** [comment]
@brettsam We don't need all identity providers test in all managed clients. We make sure every identity provider is covered in at least one managed client. MSA is covered in WinStore 8.1 e2etest, which is the purpose of this PR. The only provider not covered in any managed client e2etest is AAD. So I am going to open an issue for tracking that.

**Pull request merged**